docs/infrastructure/feature branches jenkins.md

```diff
@@ -0,0 +1,50 @@
+# Jenkins in Feature Branches
+This document describes the process for setting up CI on a feature branch of roslyn.
+
+## Push the branch
+The first step is to create the branch seeded with the initial change on roslyn. This branch should have the name `features/<feature name>`. For example: `features/mono` for working on mono work. 
+
+Assuming the branch should start with the contents of `master` the branch can be created by doing the following:
+
+``` cmd
+> git fetch origin
```

Comment:
Note that most people here have 'origin' pointing their fork, and 'upstream' pointing to dotnet/master. Further, many have pushes directly to upstream disabled. You do call out the 'Note' below but you might want to put the note first.

(I had to help somebody through this last week...)